sql2.txt

```diff
@@ -2,44 +2,44 @@
-
+SELECT products.ProductName, SUM(OrderDetails.Quantity) AS TotalCount  FROM OrderDetails JOIN products ON OrderDetails.ProductID = products.ProductID GROUP BY OrderDetails.ProductID ORDER BY ProductName;
 
 2. Retrieve the customer name and OrderDetailsID for all the OrderDetails for each customer,
 ordered by customer name. Paste your SQL statement below. You will have to join the Customers
-table with the Orders table, and the Orders table with the OrderDetails table, 
+table with the Orders table, and the Orders table wOith the OrderDetails table, 
 because the Orders table is the one that ties the Customer table to the OrderDetails table.
-
+SELECT Customers.customerName,  OrderDetails.OrderDetailID FROM Customers JOIN  OrderDetails ON Customers.CustomerID = Orders.CustomerID JOIN Orders ON Orders.OrderID = OrderDetails.OrderID ORDER BY Customers.customerName;
 
 3. Extend the previous query to join the Products table.  You are now joining the Customers, Orders,
 OrderDetails, and Products tables.  The result of your query should include the
 customer name, the product name, and the total quantity of that product ordered.  Note that a customer
 may have ordered the same product in several different orders.  So the total quantity is
 SUM(Quantity) AS TotalQuantity and you must GROUP BY CustomerID, ProductID.  Paste your
 SQL statement below.
-
+SELECT Customers.customerName, products.ProductName, SUM(OrderDetails.Quantity) AS TotalQuantity FROM Customers JOIN Orders ON Customers.CustomerID = Orders.CustomerID JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID JOIN products ON OrderDetails.ProductID = products.ProductID GROUP BY Customers.CustomerID, products.ProductID, products.ProductName;
 
 
 4. Extend the previous query to give the total price as well as the customer name, product name, 
 and total quantity.  Here you will have
 (SUM(Quantity) * Price) AS TotalPrice.  Paste your SQL statement below.
-
+SELECT Customers.customerName, products.ProductName, SUM(OrderDetails.Quantity) AS TotalQuantity, (SUM(Quantity) * Price) AS TotalPrice FROM Customers JOIN Orders ON Customers.CustomerID = Orders.CustomerID JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID JOIN products ON OrderDetails.ProductID = products.ProductID GROUP BY Customers.CustomerID, products.ProductID, products.ProductName;
 
 5. Add a new Order to the Orders table.  You will have to specify the ID of an existing customer
 to satisfy the foreign key constraint.  You will also have to specify the ID of an existing
 employee.  Paste your SQL statement below.
-
+INSERT INTO Orders(CustomerID, EmployeeID, OrderDate, ShipperID) VALUES (90, 5, "7/4/1996 ", 3); 
 
 6. Add three OrderDetails.  These should specify the ID of the new Order you just created.  For each,
 you will have to specify the ID of an existing product.  Paste your SQL statement below.
-
+INSERT INTO OrderDetails(OrderID, ProductID, Quantity) VALUES (10446, 11, 12),(10446, 11, 12),(10446, 11, 12);
```

Comment:
the value for the Three OrderDetails is not unique which will make it difficult when you want to delete one od the OrderDetails. kindly fix

Reply:
<img width="752" alt="table" src="https://github.com/mvstoyan/sqlintro/assets/95896159/c3fc10ee-cd77-461e-a2dc-57280e0ba736">
<img width="178" alt="table2" src="https://github.com/mvstoyan/sqlintro/assets/95896159/014a7a43-1222-4db4-b92a-0f26aa980dbb">
This table OrderDetails has four columns: OrderDetailsID, OrderID, ProductID, Quantity. OrderDetailsID is a unique value generated automatically.

Reply:
OrderDetailsID 524, was deleted